test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,155 @@
+package e2e_test
+
+import (
+	"context"
+	"io"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var (
+	testEnv       *envtest.Environment
+	cfg           *rest.Config
+	testUser      *envtest.AuthenticatedUser
+	ctx           context.Context
+	runtimeScheme *runtime.Scheme
+	cancel        context.CancelFunc
+	crdPath       = filepath.Join("..", "..", "config", "crd", "bases")
+)
+
+func TestE2e(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2e Suite")
+}
+
+var _ = BeforeSuite(func() {
+
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	testEnv = &envtest.Environment{
+		AttachControlPlaneOutput: false,
+	}
+
+	var err error
+
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	user := envtest.User{Name: "armada-operator-test-user", Groups: []string{"system:masters"}}
+	testUser, err = testEnv.AddUser(user, cfg)
+	Expect(err).NotTo(HaveOccurred())
+
+	runtimeScheme = scheme.Scheme
+
+	k8sClient, err := client.New(cfg, client.Options{Scheme: runtimeScheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "armada",
+		},
+	}
+
+	Expect(k8sClient.Create(ctx, ns)).To(Succeed())
+})
+
+var _ = Describe("E2e", func() {
+
+	It("Install Bases CRDs", func() {
```

Comment:
I'd check to make sure the operator is running before you test other items.  If the operator or webhooks are not working then all these tests will fail.